src/logic/projectSuccessfullySaved.js

```diff
@@ -0,0 +1,18 @@
+import {createLogic} from 'redux-logic';
+
+import {
+  showSaveIndicator,
+  hideSaveIndicator,
+} from '../actions/ui';
+
+export default createLogic({
+  type: 'PROJECT_SUCCESSFULLY_SAVED',
+  debounce: 1000,
+  async process(data, dispatch, done) {
+    dispatch(showSaveIndicator());
+    setTimeout(() => {
```

Comment:
setTimeout this way worked better with the tests as opposed to waiting for the promise to resolve

Reply:
Hmm, so I don’t think this will give us incorrect behavior, since all `redux-logic` cares about is the `done()` callback in this case, but I think `async` functions should generally only resolve when they have actually done their work, which is not the case here. What was the problem with the tests?

Reply:
Looks like this was the issue - https://stackoverflow.com/questions/52177631/jest-timer-and-promise-dont-work-well-settimeout-and-async-function

Reply:
I will play around with this

Reply:
Hey, looks like this will do it:

```diff
diff --git a/package.json b/package.json
index 9ca8782..e3a7863 100644
--- a/package.json
+++ b/package.json
@@ -197,6 +197,7 @@
     "chrono-node": "^1.3.11",
     "classnames": "^2.2.6",
     "css": "^2.2.4",
+    "delay": "^4.3.0",
     "enum": "^2.5.0",
     "enumify": "^1.0.4",
     "es6-set": "^0.1.4",
diff --git a/src/logic/__tests__/projectSuccessfullySavedTest.js b/src/logic/__tests__/projectSuccessfullySavedTest.js
index 02a9eaa..13dc346 100644
--- a/src/logic/__tests__/projectSuccessfullySavedTest.js
+++ b/src/logic/__tests__/projectSuccessfullySavedTest.js
@@ -8,12 +8,13 @@ import {
 test('should show project saved successfully', async() => {
   jest.useFakeTimers();
   const dispatch = jest.fn();
-  const done = jest.fn();
   const data = {};
-  projectSuccessfullySaved.process(data, dispatch, done);
+  const logicDone = new Promise(
+    resolve => projectSuccessfullySaved.process(data, dispatch, resolve),
+  );
   expect(dispatch).toHaveBeenCalledWith(showSaveIndicator());
   expect(dispatch).not.toHaveBeenCalledWith(hideSaveIndicator());
   jest.advanceTimersByTime(1000);
+  await logicDone;
   expect(dispatch).toHaveBeenCalledWith(hideSaveIndicator());
-  expect(done).toHaveBeenCalledWith();
 });
diff --git a/src/logic/projectSuccessfullySaved.js b/src/logic/projectSuccessfullySaved.js
index 46d2727..4e5e9ad 100644
--- a/src/logic/projectSuccessfullySaved.js
+++ b/src/logic/projectSuccessfullySaved.js
@@ -1,4 +1,5 @@
 import {createLogic} from 'redux-logic';
+import delay from 'delay';
 
 import {
   showSaveIndicator,
@@ -10,9 +11,8 @@ export default createLogic({
   debounce: 1000,
   async process(data, dispatch, done) {
     dispatch(showSaveIndicator());
-    setTimeout(() => {
-      dispatch(hideSaveIndicator());
-      done();
-    }, 1000);
+    await delay(1000);
+    dispatch(hideSaveIndicator());
+    done();
   },
 });
```

(The key is to await the `done()` callback getting called in the test)